### PR TITLE
RFC-0 Proposal for a basic CRUD API, includes a table for answers

### LINKED ISSUE #49: Use a basic database to handle answers
Define the best database engine to the use case.
Define the structure of data.

### PR DESCRIPTION
**Pull request by juanjosegzl**
Let's discuss all it has to be clarified and put it on this document.

This is the basis of what has to be implement for the sprint on Feb 21

Closes #49

### CODE REVIEW

File: rfc/0-api-draft.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+| Field  | Type  |
+| ------------ | ------------ |
+| id  | autoincrement integer Primary Key  |
+| answer  | text |
+| metadata  | text  |
+| created  | datetime  |
+| modified  | datetime  |
+| author_original  | str (save the auth id, maybe google)  |
+| author_latest | str |
+| deleted | bool |
```
Comment (hugoaguirre) [inline rfc/0-api-draft.md:18]: Should we have a previous stored answer? Just in case we mess up?

File: rfc/0-api-draft.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## Database
+
+Table: `answers`
+
+| Field  | Type  |
+| ------------ | ------------ |
+| id  | autoincrement integer Primary Key  |
+| answer  | text |
+| metadata  | text  |
```
Comment (hugoaguirre) [inline rfc/0-api-draft.md:13]: What's the purpose of this metadata field?

File: rfc/0-api-draft.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+| created  | datetime  |
+| modified  | datetime  |
+| author_original  | str (save the auth id, maybe google)  |
+| author_latest | str |
+| deleted | bool |
+
```
Comment (hugoaguirre) [inline rfc/0-api-draft.md:19]: Should we also have the tags for each answer? Maybe in the future we can query these answer by topic/tag.

**Review by hugoaguirre — COMMENTED** [review]
Looks good, minor comments added.

File: rfc/0-api-draft.md (hunk 1 above)
Comment (juanjosegzl) [reply rfc/0-api-draft.md:18]: yes it is a good idea 👍

File: rfc/0-api-draft.md (hunk 2 above)
Comment (juanjosegzl) [reply rfc/0-api-draft.md:13]: the metadata is all that additional information to help the bot find the correct answer. Like synonyms, acronyms and related words that not necessarily appear on the answer.

File: rfc/0-api-draft.md (hunk 3 above)
Comment (hugoaguirre) [reply rfc/0-api-draft.md:19]: I think you answered this question on the **metadata** answer above.

Comment (juanjosegzl) [reply rfc/0-api-draft.md:19]: To be honest I would like to discuss more about this before implementing it. I find it useful but I am not sure about the time and scope of the current milestone cc @isuscbrmid

File: rfc/0-api-draft.md (hunk 1 above)
Comment (juanjosegzl) [reply rfc/0-api-draft.md:20]: I added the field

**Review by hugoaguirre — APPROVED** [review]

**Review by fervargas94 — APPROVED** [review]

File: rfc/0-api-draft.md (hunk 1 above)
Comment (rpadro) [reply rfc/0-api-draft.md:20]: What about instead of just having the one before, It is maybe better to have an array with the history of the question (?)

**Pull request merged**